server/testdriver/core/src/main/java/org/infinispan/server/test/core/TestClient.java

```diff
@@ -99,6 +104,10 @@ public void setMethodName(String methodName) {
    }
 
    public void clearResources() {
+      if (cacheList != null) {
+         cacheList.forEach(c -> hotrod().createRemoteCacheManager().administration().removeCache(c));
```

Comment:
@ryanemerson So this is the most primitive solution which only considers caches created by HotRod client. REST client would probably require it's own list as I assume there may be tests that have only REST endpoint enabled and thus clean up via HotRod would fail?

We could also add some interface and defer the implementation of the removal to individual Administrative endpoints (HotRod and REST). Either way it requires bigger code intervention then I'd expect from this enhancement.

Reply:
We can make this work with REST by updating the `RestTestClientDriver#create` to also register caches with the TestClient. If we make `cacheList` a `Set<String>` we also don't have to worry about duplicate cache names if a test uses both the HR and REST client.

Reply:
So this solution can't be used. I have to store cache name along the instance of RestClient/RemoteCacheManager that created the cache otherwise it'll fail for auth(z) tests (I can't create a new RemoteCacheManager to delete the cache if I don't know the configuration used in the test).

Tbh. the solution isn't exactly nice as we are now storing RestClient/RemoteCacheManagers pointers in Resource List and cache maps. And if some of the tests decides to delete the cache on it's own for testing purposes the cleanup part will fail. But anything other solution would require bigger changes. I feel we are overkilling the solution here a bit for few tests that are affected (few tests I know of)

Reply:
I think storing the map is fine as we're clearing the resources after each test method.

> And if some of the tests decides to delete the cache on it's own for testing purposes the cleanup part will fail.

Is an exception thrown by HR/rest client when a remove is not possible due to the cache already being deleted? If so, we should catch the exception, log the issue and proceed with the loop.